### PR TITLE
Scrolling view keeps selections

### LINKED ISSUE #917: Scrolling view using view mode or mouse removes additional cursors
### Reproduction steps

1. Create additional cursors.
2. Scroll the view using `zj`, `zk`, `zu`, `zd`, `zb`, `zf`, or mouse scroll (`zt`, `zz`, and `zb` do not cause the issue).
3. Additional cursors/selections other than primary will disappear.

### PR DESCRIPTION
**Pull request by k2d222**
Fixes #917.

* Scrolling will never change secondary selections.
* Primary selection is replaced with a single char selection when the cursor gets out of view.

This is the same behavior as in Kakoune. I believe in vim / nvim it would grow / shrink the main selection instead of deleting it.

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -1282,16 +1282,23 @@ pub fn scroll(cx: &mut Context, offset: usize, direction: Direction) {
...
+        let anchor = if doc.mode == Mode::Select {
+            range.anchor
+        } else {
+            head
+        };
 
-    // TODO: only manipulate main selection
-    doc.set_selection(view.id, Selection::single(anchor, head));
+        // replace primary selection with an empty selection at cursor pos
+        let prim_sel = Range::new(anchor, head);
```
Comment (dead10ck) [inline helix-term/src/commands.rs:1296]: Do we need to clone, or is it possible just to mutate the primary range that we already got from above?

**Review by dead10ck — CHANGES_REQUESTED** [review]
Nice, thank you!

File: helix-term/src/commands.rs (hunk 1 above)
Comment (k2d222) [reply helix-term/src/commands.rs:1296]: Selections are currently not mutably accessible outside Document. I believe this protects the Integrity of the selections, e.g. not having overlapping selections (see: https://github.com/helix-editor/helix/blob/aaa42e1a69e4bf88e6761e51085957201295ca52/helix-core/src/selection.rs#L523)

Ranges and Selections are pretty lightweight too: Range is 3 `usize` and Selection is 1 vec of Ranges + 1 `usize`

Comment (dead10ck) [reply helix-term/src/commands.rs:1296]: Yeah, but `Range: Copy`, and `Selection::primary` returns an owned Range, which is its own copy, if I'm not mistaken.

Comment (k2d222) [reply helix-term/src/commands.rs:1296]: I'm not sure to get your point. Currently, `Document::set_selection` is the only way to modify selections and it asks for a owned `Selection`, so a copy of Selection is required anyway. The Ranges inside Selections are not mutable in any way so a copy or a new range is also required.

Comment (dead10ck) [reply helix-term/src/commands.rs:1296]: Sorry, I see what you mean now: `Selection::replace` takes a `mut self`, so cloning is necessary. Never mind me.

**Comment by archseer** [comment]
Thanks! 🎉

**Pull request merged**